app/screens/send-bitcoin-screen/use-fee.ts

```diff
@@ -0,0 +1,213 @@
+import { useState } from "react"
+import { gql, useApolloClient, useMutation, useQuery } from "@apollo/client"
+
+import { textCurrencyFormatting } from "../../utils/currencyConversion"
+import { getMyUsername, USER_WALLET_ID } from "../../graphql/query"
+
+const LIGHTNING_FEES = gql`
+  mutation lnInvoiceFeeProbe($input: LnInvoiceFeeProbeInput!) {
+    lnInvoiceFeeProbe(input: $input) {
+      errors {
+        message
+      }
+
+      amount
+    }
+  }
+`
+
+const NO_AMOUNT_LIGHTNING_FEES = gql`
+  mutation lnNoAmountInvoiceFeeProbe($input: LnNoAmountInvoiceFeeProbeInput!) {
+    lnNoAmountInvoiceFeeProbe(input: $input) {
+      errors {
+        message
+      }
+
+      amount
+    }
+  }
+`
+
+const ONCHAIN_FEES = gql`
+  query onChainTxFee(
+    $walletId: WalletId!
+    $address: OnChainAddress!
+    $amount: SatAmount!
+    $targetConfirmations: TargetConfirmations
+  ) {
+    onChainTxFee(
+      walletId: $walletId
+      address: $address
+      amount: $amount
+      targetConfirmations: $targetConfirmations
+    ) {
+      amount
+      targetConfirmations
+    }
+  }
+`
+
+type FeeType = {
+  value: number | null
+  status: "loading" | "error" | "unset" | "set"
+}
+
+type UseFeeInput = {
+  address: string
+  amountless: boolean
+  invoice: string
+  paymentType: string
+  sameNode: boolean
+  paymentSatAmount: number
+  btcPrice: number
+  primaryCurrency: CurrencyType
+}
+
+type UseFeeReturn = {
+  value: number | null
+  status: "loading" | "error" | "unset" | "set"
+  text: string
+}
+
+const useFee = ({
+  address,
+  amountless,
+  invoice,
+  paymentType,
+  sameNode,
+  paymentSatAmount,
+  btcPrice,
+  primaryCurrency,
+}: UseFeeInput): UseFeeReturn => {
+  const client = useApolloClient()
+  const { data: dataUserWalletId } = useQuery(USER_WALLET_ID, {
+    variables: { username: getMyUsername(client) },
+  })
+  const walletId = dataUserWalletId?.userWalletId ?? ""
+
+  const [fee, setFee] = useState<FeeType>({
+    value: null, // TODO: get rid of this
```

Comment:
Not sure what this TODO means.  Can we get rid of it now?  If not then could we add the criteria for getting rid of it in the TODO?